[WIP] schemachanger: flesh out zone config elements

postamar opened the pull request:
Previously, the declarative schema changer modeled zone configurations only via placeholder elements. This commit fleshes out the element model to include all zone configuration entries as distinct elements keyed by the name of the configuration key.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/103806)
<!-- Reviewable:end -->

postamar commented:
Marked as WIP because none of this actually gets exercised yet.

Commit:
schemachanger: flesh out zone config elements

Previously, the declarative schema changer modeled zone configurations
only via placeholder elements. This commit fleshes out the element model
to include all zone configuration entries as distinct elements keyed by
the name of the configuration key.

Release note: None

postamar commented:
Closing this. cc @fqazi and @Xiang-Gu in case you want to pick this up sometime. I didn't get particularly far, though.
